src/plugins/enzymes/src/FindEnzymesDialog.cpp

```diff
@@ -589,6 +635,57 @@ void FindEnzymesDialog::accept() {
     QDialog::accept();
 }
 
+void FindEnzymesDialog::sl_setEnzymesBySuppliers(QStringList checkedSuppliers) {
+    const auto enzymes = EnzymesSelectorWidget::getLoadedEnzymes();
+    QList<SEnzymeData> visibleEnzymes;
+    for (const auto& enzyme : qAsConst(enzymes)) {
+        for (const auto& supplier : qAsConst(enzyme->suppliers)) {
+            CHECK_CONTINUE(checkedSuppliers.contains(supplier));
+
+            visibleEnzymes.append(enzyme);
+            break;
+
+        }
+    }
+    enzSel->setEnzymesList(visibleEnzymes);
+    AppContext::getSettings()->setValue(EnzymeSettings::CHECKED_SUPPLIERS, checkedSuppliers.join(SUPPLIERS_LIST_SEPARATOR));
+}
+
+void FindEnzymesDialog::sl_updateSuppliers() {
+    const auto& loadedSuppliers = EnzymesSelectorWidget::getLoadedSuppliers();
+    cbSuppliers->clear();
+    for (const auto& supplier : qAsConst(loadedSuppliers)) {
+        cbSuppliers->addItem(supplier);
+    }
+    QString selStr = AppContext::getSettings()->getValue(EnzymeSettings::CHECKED_SUPPLIERS).toString();
+    if (selStr.isEmpty()) {
+        cbSuppliers->setCheckedItems(loadedSuppliers);
+    } else {
+        cbSuppliers->setCheckedItems(selStr.split(SUPPLIERS_LIST_SEPARATOR));
```

Comment:
You should either avoid translation of default ("Public sources," "Not defined") supplier or save it separately. Otherwise if a user switches language the setting is lost.